plugwise/helper.py

```diff
@@ -469,7 +469,9 @@ def _energy_device_info_finder(self, appliance: etree, appl: Munch) -> Munch:
                 appl.model = version_to_model(hw_version)
             appl.fw = module_data["firmware_version"]
             appl.zigbee_mac = module_data["zigbee_mac_address"]
-            return appl
+            # Filter appliance without zigbee_mac, it's an orphaned device
+            if appl.zigbee_mac:
```

Comment:
Now we don't return (i.e. continue with the other if statements) ... shouldn't we `if not: return` and leave the generic return? (i.e. we probably hit L561 or something now)

Reply:
Good point. We should always `return appl`, that's what the calling function expects. Implemented.

Reply:
And, continuing with the other if statement is not needed. They are mutually exclusive.